src/megaclient.cpp

```diff
@@ -278,10 +292,14 @@ void MegaClient::mergenewshare(NewShare *s, bool notify)
                             {
                                 // This is currently a pending share that needs to be upgraded to a full share
                                 // erase from pending shares & delete the pending share list if needed
-                                if (n->pendingshares->erase(s->pending) && notify)
+                                if (n->pendingshares->erase(s->pending))
                                 {
-                                    n->changed.pendingshares = true;
-                                    notifynode(n);
+                                    if (notify)
+                                    {
+                                        n->changed.pendingshares = true;
+                                        notifynode(n);
+                                    }
+                                    delete *sharep;
```

Comment:
I agree. Probably, the `if (n->pendingshares->erase(s->pending))` is not required at all, since `*sharep` is assigned to an existing value in the map and checked in the previous `if`, so the `erase()` won't fail ever.